src/time.rs

```diff
@@ -4,28 +4,26 @@ use core::ops::{Add, AddAssign, Sub, SubAssign};
 
 /// Kernel start time, measured in (seconds, nanoseconds) since Unix epoch
 pub static START: RwLock<(u64, u32)> = RwLock::new((0, 0));
-/// Kernel up time, measured in (seconds, nanoseconds) since `time::START`
-pub static OFFSET: RwLock<(u64, u32)> = RwLock::new((0, 0));
 
 /// Return the start time of the kernel
 pub fn start() -> SystemTime {
     let (secs, nanos) = *START.read();
     SystemTime(Duration::new(secs, nanos))
 }
 
-/// Return the up time of the kernel
+/// Return the up time of the kernel in nanoseconds
 #[inline]
-fn monotonic() -> (u64, u32) {
-    *OFFSET.read()
+fn monotonic() -> u64 {
+    ::arch::x64::devices::high_precision_timer::now()

```

Comment:
I debated importing/not importing this.

Once we support more architectures, we're going to have to choose which function we are calling based on a cfg flag in here. For that use case it seemed simplest to have everything on one line. For other purposes it's pretty much irrelevant.

Reply:
We'll likely have a `high_precision_timer` on every arch, so I think it's safe to just do `arch::devices::high_precision_timer` or to import it.

Reply:
Ok, changed

Reply:
You can just do `use arch::devices::high_precision_timer;`

Reply:
That's... what I just did.

Reply:
Oh... cool. Will update it.